### PR TITLE
export XPUStream to doc

### PR DESCRIPTION
**Pull request by guangyey**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #121398

# Motivation
We would like to export XPUStream to public [doc](https://pytorch.org/cppdocs/api/library_root.html). The detailed documentation can help users understand and utilize XPU more effectively.

# Additional Context
A detailed XPUStream API and usage should be documented to public doc, like cuda's [doc](https://github.com/pytorch/pytorch/blob/main/docs/cpp/source/notes/tensor_cuda_stream.rst).

cc @svekars @brycebortree @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

**Review by albanD — COMMENTED** [review]
That sounds fine. I would say that we have very very few users of our c++ API.
I don't see this on the rendered doc: https://docs-preview.pytorch.org/pytorch/pytorch/121398/cppdocs/search.html?q=xpustream&check_keywords=yes&area=default  (you can find this link in the drci comment)

**Review by gujinghui — APPROVED** [review]

**Comment by gujinghui** [comment]
> That sounds fine. I would say that we have very very few users of our c++ API. I don't see this on the rendered doc: https://docs-preview.pytorch.org/pytorch/pytorch/121398/cppdocs/search.html?q=xpustream&check_keywords=yes&area=default (you can find this link in the drci comment)

@albanD  PyTorch has some C++ users, perhaps they are not very active in online community.  :)

File: c10/xpu/XPUStream.h [hunk 1]
```diff
@@ -30,21 +30,19 @@ static constexpr int max_compile_time_stream_priorities = 2;
 
 /*
  * This serves as a wrapper around c10::Stream and acts as a representation for
- * a SYCL queue. On each device, a SYCL queue pool consists of kStreamsPerPool
- * queues, and you can access a particular queue by its index. The index is
- * extracted from XPUStream.id().
+ * a SYCL queue, which allows to asynchronously execute tasks on XPU.
```
Comment (jgong5) [inline c10/xpu/XPUStream.h:33]:
```suggestion
 * a SYCL queue, which allows asynchronous execution of XPU tasks.
```

**Review by jgong5 — APPROVED** [review]
some nits, otherwise LGTM.

File: c10/xpu/XPUStream.h (hunk 1 above)
Comment (guangyey) [reply c10/xpu/XPUStream.h]: Thanks, done.

**Comment by albanD** [comment]
cc @svekars do you know why the Library API does not render on the preview? How do we fix that?

**Comment by guangyey** [comment]
@svekars Could you please help review this RP?

**Comment by guangyey** [comment]
@svekars Could you help review this PR?

**Review by EikanWang — APPROVED** [review]

**Comment by guangyey** [comment]
Hi @svekars and @albanD , what can I do for this PR?

**Comment by guangyey** [comment]
@svekars it would be appreciated if you could help review this PR.

**Comment by guangyey** [comment]
@svekars may I know if you have comments on this PR?

**Review by albanD — APPROVED** [review]
Stamping to unblock. Will ping @svekars directly if she has other comments.

**Comment by guangyey** [comment]
> Stamping to unblock. Will ping @svekars directly if she has other comments.

Thanks for your help!

**Pull request closed without merging**